atlasdb-feign/src/main/java/com/palantir/atlasdb/http/PollingRefreshable.java

```diff
@@ -32,11 +33,13 @@
 
 /**
  * A PollingRefreshable serves as a bridge between a {@link Supplier} and {@link Refreshable}, polling for changes
- * in the value of the Supplier and, if detecting a change, pushing it to the linked Refreshable.
+ * in the value of the Supplier and, if detecting a change, pushing it to the linked Refreshable, after possibly
+ * applying a transform to the value.
```

Comment:
I'd suggest adding a note here on what the semantics of transforms are. Specifically, the transform is computed once and its result is cached - so even if the transform somehow changes such that its output on the same input value from the supplier changes, new values will _not_ be propagated in that case.

Reply:
i.e. it needs to be a pure fuction? 😁

Reply:
Yeah, it should be a pure function - or if it's not, then the user should understand that it may not be recalculated!

Reply:
Some non-pure functions would be ok if their state only evolves as a result of changes in the input value, though I can't think of any significant examples beyond contrived ones e.g. a function that ignores the input value, other than counting the number of occasions on which we saw that it changed.